### DISCUSSION TITLE
Best way looping through pixels or bands using C++

### DISCUSSION THREAD

**Discussion by ademmler** [libvips/libvips#2029] [Q&A]
Hi,

what is the best way for looping through pixels or bands using C++.
I want to modify (doing math on each pixel) of multichannel images.

Thx Alex

**Comment by jcupitt** [comment]
Hello @ademmler,

The C++ binding has a full set of operator overloads, so you can just write `a + b << 2 & c` or whatever.

**Comment by jcupitt** [comment]
... in case that wasn't clear, something like:

```C++
VImage x =  VImage::new_from_file (filename);
x = ((x * 23 + 12) & (x >> 3)).sin();
```

You can use array subscript to address bands.
